walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -12,12 +12,133 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
+import static java.lang.Math.toIntExact;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {    
+    // not sure why this works!! but I have to cast the request 
+    //of attendees as an int 
+    Collection<String> attendeesCollection = request.getAttendees();
+    Collection<String> optionalAttendeesCollection = request.getOptionalAttendees();
+
+    List<TimeRange> badTimes = new ArrayList<TimeRange>();
+    List<TimeRange> goodTimes = new ArrayList<TimeRange>();
+
+    // get the duration of the event, haven't implemented yet
+    long durationLong = request.getDuration();
+    int duration = (int) durationLong;
+
+    if(duration >= TimeRange.WHOLE_DAY.duration())
+    {
+        return Arrays.asList();
+    }
+
+    Set<String> participants = new HashSet<String>();
+    if(!(request.getOptionalAttendees().isEmpty()))
+    {   
+        for(String attendee: optionalAttendeesCollection)
+        {
+            participants.add(attendee);
+        }
+    }
+
+    for(String attendee: attendeesCollection)
+    {
+        participants.add(attendee);
+    }
+
+
+
+    // before you go any further, I have really bad runtime on here!
+    // sorry! :(
+    //getting each attendee
+
+    for(Event thisMeeting: events) {
+        // get the attendees for this meeting (all meeting participants)
+        Set<String> amp = new HashSet<String>(thisMeeting.getAttendees());
+        // for every attendee in this meeting
+        for(String participant1: amp)
+        {
+            // for every attendee in the meeting we want
+            for(String participant: participants)
+            {
+                // if the participant is in this meeting, add it 
+                // to the bad times we collect
+                if(participant == participant1)
+                {
+                    // if there is a participant who has a meeting that's the whole day,
+                    // then there will be no good meeting times 
+                    if((thisMeeting.getWhen().equals(TimeRange.WHOLE_DAY)))
+                    {
+                        return Arrays.asList();
+                    }
+                    badTimes.add(thisMeeting.getWhen());
+                }
+            }
+        }
+    }
+
+    //if there are no bad times, return the whole day
+    if(badTimes.size() == 0)
+    {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // if there's one event, schedule good times 
```

Comment:
Please fit the comment to a single line.

Reply:
Zu, do you know if there's a way to check the # of characters on a line in this IDE? I've been eyeballing it most the time so I thought I'd ask